adapters/repos/db/vector/hnsw/compress.go

```diff
@@ -20,51 +20,56 @@ import (
 	ent "github.com/weaviate/weaviate/entities/vectorindex/hnsw"
 )
 
-func (h *hnsw) Compress(cfg ent.PQConfig) error {
-	h.shardedNodeLocks.RLock(0)
-	node := h.nodes[0]
-	h.shardedNodeLocks.RUnlock(0)
+func (h *hnsw) Compress(cfg ent.UserConfig) error {
+	h.compressActionLock.Lock()
+	defer h.compressActionLock.Unlock()
+	data := h.cache.All()
+	if cfg.PQ.Enabled {
+		h.shardedNodeLocks.RLock(0)
+		node := h.nodes[0]
+		h.shardedNodeLocks.RUnlock(0)
```

Comment:
could you please merge master into `HNSW_BQ` branch. I believe is changed already